### PR TITLE
NGSTACK-480: implement access to relation Locations

### PR DESCRIPTION
**Pull request by pspanja**
This implements access to relation Locations.

Note that this does not implement real Location relations, related Locations are are **visible** **main** Locations of the related Content, and relations are internally still implemented on Content level.

Two new methods are added to the `RelationService`:
- `loadFieldRelationLocation()`
- `loadFieldRelationLocations()`

Three new methods are added to the `Content` object:
- `getFieldRelationLocation()`
- `getFieldRelationLocations()`
- `filterFieldRelationLocations()`

Four new query types are implemented:
- `SiteAPI:Location/Relations/AllTagFields`
- `SiteAPI:Location/Relations/ForwardFields`
- `SiteAPI:Location/Relations/ReverseFields`
- `SiteAPI:Location/Relations/TagFields`

### CODE REVIEW

**Review by hknezevic — APPROVED** [review]
@pspanja looks good!

**Comment by emodric** [comment]
Agreed, looks good :+1:

File: lib/API/RelationService.php [hunk 1]
```diff
@@ -50,4 +51,42 @@ public function loadFieldRelations(
...
+     *
+     * Note: only visible main Location of the related Content will be used.
+     *
+     * @param \Netgen\EzPlatformSiteApi\API\Values\Content $content
+     * @param string $fieldDefinitionIdentifier
+     * @param array $contentTypeIdentifiers
+     *
+     * @return null|\Netgen\EzPlatformSiteApi\API\Values\Location
+     */
+    public function loadLocationFieldRelation(
```
Comment (iherak) [inline lib/API/RelationService.php]: Naming nitpick:
I'd prefer this to be named `loadFieldRelationLocation` as we are loading location(s) from the field relation.
Current naming implies that we are loading field relation from the location, which is not true.

Same for all other corresponding methods.

I am not sure if I am making sense here, so feel free to ask for more explanations :)

Comment (hknezevic) [reply lib/API/RelationService.php]: @iherak Actually, now that you pointed it out, I agree it does make more sense with the naming pattern you proposed. I didn't give it much thought at the first glance.
So the desired format change would be:
loadLocationFieldRelation(s) -> loadFieldRelationLocation(s)
filterLocationFieldRelation(s) -> filterFieldRelationLocation(s)
cc @pspanja

Comment (pspanja) [reply lib/API/RelationService.php]: Agreed, that's much better 👍

Comment (pspanja) [reply lib/API/RelationService.php]: Updated in https://github.com/netgen/ezplatform-site-api/pull/190/commits/8a63631cddb0d5b502f1296da922012cd9b45c86

**Review by iherak — APPROVED** [review]
Looks great now!

**Comment by pspanja** [comment]
Thanks for the review!

**Pull request merged**